torch/csrc/cuda/CUDAPluggableAllocator.cpp

```diff
@@ -104,7 +104,10 @@ void* CUDAPluggableAllocator::malloc(
   void* r = alloc_fn_(size, device, stream);
   {
     const std::lock_guard<std::mutex> lock(allocator_mutex_);
-    allocation_metadata_.emplace(r, _AllocationMetadata(size, device, stream));
+    auto result = allocation_metadata_.try_emplace(r, size, device, stream);
```

Comment:
How can previous allocation with the same address not have been erased during `free`?

Reply:
SO that's the original sin, `raw_delete` behavior is different from context deleter. Context deleter has to also clean up metadata.

Reply:
Ok, then not recording the metadata sounds like a better way.

Reply:
So essentially the context as implemented in #130472 is bc-breaking? Nice! If we stop recording metadata, that is a bc-breaking change. If we stop cleaning up metadata, metadata is useless. cc @syed-ahmed